lib/node_modules/@stdlib/ndarray/base/mskfilter/lib/main.js

```diff
@@ -0,0 +1,356 @@
+	blockedmskfilter2d,
+	blockedmskfilter3d,
+	blockedmskfilter4d,
+	blockedmskfilter5d,
+	blockedmskfilter6d,
+	blockedmskfilter7d,
+	blockedmskfilter8d,
+	blockedmskfilter9d,
+	blockedmskfilter10d
+];
+
+var BLOCKED_ACCESSOR_MSKFILTER = [
+	blockedaccessormskfilter2d,
+	blockedaccessormskfilter3d,
+	blockedaccessormskfilter4d,
+	blockedaccessormskfilter5d,
+	blockedaccessormskfilter6d,
+	blockedaccessormskfilter7d,
+	blockedaccessormskfilter8d,
+	blockedaccessormskfilter9d,
+	blockedaccessormskfilter10d
+];
+
+var MAX_DIMS = MSKFILTER.length - 1;
+
+
+// MAIN //
+
+/**
+* Applies a mask to a provided input ndarray and assigns unmasked values to elements in a provided one-dimensional output ndarray.
+*
+* ## Notes
+*
+* -   Each provided ndarray should be an `object` with the following properties:
+*
+*     -   **dtype**: data type.
+*     -   **data**: data buffer.
+*     -   **shape**: dimensions.
+*     -   **strides**: stride lengths.
+*     -   **offset**: index offset.
+*     -   **order**: specifies whether an ndarray is row-major (C-style) or column major (Fortran-style).
+*
+* @param {ArrayLikeObject<Object>} arrays - array-like object containing input array, mask array and output array
+* @throws {Error} input and mask arrays must have the same shape
+* @throws {Error} output array must have the one dimensional shape
+* @throws {Error} input and output arrays must have the same data type
+* @returns {void}
+*
+* @example
+* var Float64Array = require( '@stdlib/array/float64' );
+* var Uint8Array = require( '@stdlib/array/uint8' );
+*
+* // Define the shape of the input and mask arrays:
+* var shape = [ 3, 1, 2 ];
+*
+* // Define the array strides:
+* var sx = [ 2, 2, 1 ];
+* var sm = [ 2, 2, 1 ];
+* var sy = [ 1 ];
+*
+* // Define the index offsets:
+* var ox = 1;
+* var om = 0;
+*
+* // Create the input, mask and output ndarray-like objects:
+* var x = {
+*     'dtype': 'float64',
+*     'data': new Float64Array( [ 1.0, 2.0, 3.0, 4.0, 5.0, 6.0, 7.0, 8.0, 9.0, 10.0, 11.0, 12.0 ] ),
+*     'shape': shape,
+*     'strides': sx,
+*     'offset': ox,
+*     'order': 'row-major'
+* };
+* var mask = {
+*     'dtype': 'uint8',
+*     'data': new Uint8Array( [ 1, 0, 1, 0, 1, 0 ] ),
+*     'shape': shape,
+*     'strides': sm,
+*     'offset': om,
+*     'order': 'row-major'
+* };
+* var out = {
+*     'dtype': 'float64',
+*     'data': new Float64Array( 3 ),
+*     'shape': [ 3 ],
+*     'strides': sy,
+*     'offset': 0,
+*     'order': 'row-major'
+* };
+*
+* mskfilter( [ x, mask, out ] );
+*
+* console.log( out.data );
+* // => <Float64Array>[ 2.0, 4.0, 6.0 ]
+*/
+function mskfilter( arrays ) {
+	var ndims;
+	var xmmv;
+	var mmmv;
+	var mask;
+	var shm;
+	var shx;
+	var iox;
+	var iom;
+	var len;
+	var ox;
+	var om;
+	var sx;
+	var sm;
+	var ns;
+	var d;
+	var x;
+	var y;
+	var i;
+
+	// Unpack the ndarrays and standardize ndarray meta data:
+	x = ndarray2object( arrays[ 0 ] );
+	mask = ndarray2object( arrays[ 1 ] );
```

Comment:
```suggestion
	if ( isBooleanArray( arrays[ 1 ] ) ) {
		mask = ndarray2object( reinterpretBoolean( arrays[ 1 ], 0 ) );
	} else {
		mask = ndarray2object( arrays[ 1 ] );
	}
```

Reply:
We can avoid the accessor protocol loops when provided a boolean array, as it can be reinterpreted as a `Uint8Array` for the purposes of this function.

We cannot do the same for complex number arrays.